src/cc/api/BPF.h

```diff
@@ -263,7 +263,7 @@ class BPF {
   int poll_perf_buffer(const std::string& name, int timeout_ms = -1);
 
   StatusTuple load_func(const std::string& func_name, enum bpf_prog_type type,
-                        int& fd, unsigned flags = 0);
+                        int& fd, unsigned flags = 0, enum bpf_attach_type = (bpf_attach_type) MAX_BPF_ATTACH_TYPE);
```

Comment:
I think that using MAX_BPF_ATTACH_TYPE here might not the best option. Note that MAX_BPF_ATTACH_TYPE equal to the last ATTACH_TYPE + 1. Say it is 38. If in the future, a new attach type is added, the MAX_BPF_ATTACH_TYPE becomes 39.

So if an old bcc package may not work properly with newer kernel. So I think the default value as -1 probably a better choice.